library/core/src/alloc/layout.rs

```diff
@@ -321,7 +326,10 @@ impl Layout {
         let alloc_size = padded_size.checked_mul(n).ok_or(LayoutError)?;
 
         // The safe constructor is called here to enforce the isize size limit.
-        Layout::from_size_alignment(alloc_size, self.align).map(|layout| (layout, padded_size))
+        match Layout::from_size_alignment(alloc_size, self.align) {
```

Comment:
Hmm, moving off `map` for `const` is fine, but is there something else it could use to still be shorter than the full `match`?  I think `?` works (unstably) in `const fn` now (right?) so maybe that?